src/plugins/formulas/engine/register.js

```diff
@@ -0,0 +1,186 @@
+import staticRegister from '../../../utils/staticRegister';
+import { isUndefined } from '../../../helpers/mixed';
+import { warn } from '../../../helpers/console';
+import { PLUGIN_KEY } from '../formulas';
+import { mergeEngineSettings } from './settings';
+
+/**
+ * Setups the engine instance. It either creates a new (possibly shared) engine instance, or attaches
+ * the plugin to an already-existing instance.
+ *
+ * @param {object} hotSettings Object containing the Handsontable settings.
+ * @param {string} hotId Handsontable guid.
+ * @returns {null|object} Returns the engine instance if everything worked right and `null` otherwise.
+ */
+export function setupEngine(hotSettings, hotId) {
```

Comment:
> A solution to this would be to export a static method on the plugin class, like createWorkbook(hf: HyperFormula, settings: HyperFormulaSettings), which would return a fresh and nicely configured hf instance, that can be passed into hot or used outside, with its settings all already set.

What if the workbook configuration is in conflict with the HOT config? For example,  workbook's `maxRows` < hot's `maxRows`?

> This would remove the need of using a global hyperformula instances registry, just to set some configuration 
options.

That's assuming, that the workbook would be the only way to pass HF to the plugin, right?

Reply:
> That's assuming, that the workbook would be the only way to pass HF to the plugin, right?

HF instance, yes.

> What if the workbook configuration is in conflict with the HOT config? For example, workbook's maxRows < hot's maxRows?

I'm not sure, but I think this also shows a bigger problem - we can't exactly sync `maxRows` in both hf and hot as you could have two hot instances with different `maxRows` values that are connected to a single hf instance that only has global setting. I'm not sure what we can do about it, to be honest.

Reply:
> HF instance, yes. You could still pass in the constructor so that the plugin would create a workbook for you internally.

In that case, I'd still keep the global registry for the `destroy` purposes - I know it's probably overkill with the current state of HF, but better safe than sorry ;)

> I'm not sure what we can do about it, to be honest.

Yup, me neither ;d I think it's unsolvable, honestly.

@wojciechczerniak What do you think about the workbook idea?

Reply:
> In that case, I'd still keep the global registry for the destroy purposes - I know it's probably overkill with the current state of HF, but better safe than sorry ;)

Yeah I'm already convinced about us having to `destroy` anyway, no worries haha

Reply:
I think it would break our convention across all other settings. This require a deeper discussion on our settings object. Let's not do this right now on such short notice.